services/wallet/blockchainstate_test.go

```diff
@@ -0,0 +1,38 @@
+package wallet
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func setupTestState(t *testing.T) (s *BlockChainState) {
+	return NewBlockChainState(nil, nil)
+}
+
+func TestEstimateLatestBlockNumber(t *testing.T) {
+	state := setupTestState(t)
+
+	state.setLatestBlockDataForChain(1, LatestBlockData{
+		blockNumber:   uint64(100),
+		timestamp:     time.Now().Add(-31 * time.Second),
+		blockDuration: 10 * time.Second,
+	})
+
+	state.setLatestBlockDataForChain(2, LatestBlockData{
+		blockNumber:   uint64(200),
+		timestamp:     time.Now().Add(-5 * time.Second),
```

Comment:
not super happy about the UTs depending on real time measurements...
Not sure if we've got a nice mock for the `time` module somewhere, but if not you could add a member `TimeGetFn` to `BlockChainState`, which normally gets populated with `time.Now()` and will let you override it (ie. set any arbitrary current time value) for testing purposes.